File: vivcord.client/src/app/shared/messaging/dto/message.dto.ts
export interface MessageDTO {
  id: string | number;
  senderId: string;
  senderName?: string;
  senderAvatarUrl?: string | null;
  text: string;
  status: MessageStatus;
  attachmentUrl?: string;
  attachmentType?: 'image' | 'video';
  timestamp?: string | Date;
  createdAt?: string | Date;
}

export type MessageStatus = 'sending' | 'sent' | 'error';

File: Vivcord.Server/DTO/MessageDTO.cs
namespace Vivcord.Server.DTO
{
    public record MessageDto
    {
        public int Id { get; set; }
        public Guid SenderId { get; set; }
        public Guid TargetUserId { get; set; }
        public string Text { get; set; }
        public string? AttachmentUrl { get; set; }
        public string? AttachmentType { get; set; }
    }
}
